### PR TITLE
Completed es6-classes

### PR DESCRIPTION
**Pull request by t-doan**
es6-classes-1
es6-classes-2
es6-classes-3

### CODE REVIEW

File: es6-classes/circle.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/* eslint-disable no-undef -- Shape is imported by .html */
+/* exported Circle */
+
+class Circle extends Shape {
+  constructor(radius) {
+    super(radius);
```
Comment (shawnkost) [inline es6-classes/circle.js:6]: You should be passing 2 arguments into this. One for your `perimeter`, and one for your `area`

File: es6-classes/square.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/* eslint-disable no-undef -- Shape is imported by .html */
+/* exported Square */
+
+class Square extends Shape {
+  constructor(width) {
+    super(width);
```
Comment (shawnkost) [inline es6-classes/square.js:6]: You should be passing 2 arguments into this. One for your `perimeter`, and one for your `area`

**Review by shawnkost — CHANGES_REQUESTED** [review]
Couple changes here Tai!

File: es6-classes/circle.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/* eslint-disable no-undef -- Shape is imported by .html */
+/* exported Circle */
+
+class Circle extends Shape {
+  constructor(area, perimeter, radius) {
```
Comment (shawnkost) [inline es6-classes/circle.js:5]: This should only take 1 argument for your `radius`. Utilizing your `radius`, you need to calculate what `area` & `perimeter` should be, and that's what you pass into `super`

File: es6-classes/circle.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    super(area, perimeter);
+    this.radius = radius;
+  }
+
+  describe() {
+    return `${myShape.describe()} radius: ${this.radius}`;
```
Comment (shawnkost) [inline es6-classes/circle.js:11]: This should be `super.describe()` rather than `myShape`

File: es6-classes/square.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+/* eslint-disable no-undef -- Shape is imported by .html */
+/* exported Square */
+
+class Square extends Shape {
+  constructor(area, perimeter, width) {
```
Comment (shawnkost) [inline es6-classes/square.js:5]: This should only take 1 argument for your `width`. Utilizing your `width`, you need to calculate what `area` & `perimeter` should be, and that's what you pass into `super`

File: es6-classes/square.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    super(area, perimeter);
+    this.width = width;
+  }
+
+  describe() {
+    return `${myShape.describe()} width: ${this.width}`;
```
Comment (shawnkost) [inline es6-classes/square.js:11]: This should be `super.describe()` rather than `myShape`

**Review by shawnkost — CHANGES_REQUESTED** [review]
Some more changes here Tai!

File: es6-classes/circle.js [hunk 7]
```diff
@@ -2,15 +2,15 @@
 /* exported Circle */
 
 class Circle extends Shape {
-  constructor(area, perimeter, radius) {
-    super(area, perimeter);
+  constructor(radius) {
+    super(Math.PI * (radius * radius), 2 * Math.PI * radius);
     this.radius = radius;
```
Comment (t-doan) [inline es6-classes/circle.js:7]: Fixed the changes

File: es6-classes/circle.js [hunk 8]
```diff
@@ -2,15 +2,15 @@
...
   }
 
   describe() {
-    return `${myShape.describe()} radius: ${this.radius}`;
+    return `${super.describe()} radius: ${this.radius}`;
```
Comment (t-doan) [inline es6-classes/circle.js:11]: Fixed the changes

File: es6-classes/square.js [hunk 9]
```diff
@@ -2,15 +2,15 @@
 /* exported Square */
 
 class Square extends Shape {
-  constructor(area, perimeter, width) {
-    super(area, perimeter);
+  constructor(width) {
+    super(width * width, width * 4);
```
Comment (t-doan) [inline es6-classes/square.js:6]: Fixed the changes

File: es6-classes/square.js [hunk 10]
```diff
@@ -2,15 +2,15 @@
...
     this.width = width;
   }
 
   describe() {
-    return `${myShape.describe()} width: ${this.width}`;
+    return `${super.describe()} width: ${this.width}`;
```
Comment (t-doan) [inline es6-classes/square.js:11]: Fixed the changes

**Review by shawnkost — APPROVED** [review]
👍🏻

**Pull request merged**